src/lazy/encoder/write_as_ion.rs

```diff
@@ -35,6 +37,19 @@ pub trait WriteAsIon {
     fn write_as_ion<V: AnnotatableValueWriter>(&self, writer: V) -> IonResult<()>;
 }
 
+impl WriteAsIon for &Element {
+    fn write_as_ion<V: AnnotatableValueWriter>(&self, writer: V) -> IonResult<()> {
+        let annotations = self.annotations().iter().collect::<Vec<&Symbol>>();
+        if !annotations.is_empty() {
+            self.value()
+                .write_as_ion_value(writer.with_annotations(&annotations))
+        } else {
+            self.value()
+                .write_as_ion_value(writer.without_annotations())
+        }
+    }
```

Comment:
We should add a `pub(crate)` way for this to access the `Annotations`' existing `Vec` so we don't have to `collect()` a new one. For now, you can optimize the no-annotations case a bit by doing:
```rust
if self.annotations().is_empty() {
    self.value().write_as_ion_value(writer.without_annotations())
} else {
    self.value.write_as_ion_value(writer.with_annotations(&Vec::from_iter(annotations.iter()))
}
```